src/Microsoft.TemplateEngine.Orchestrator.RunnableProjects/RunnableProjectGenerator.cs

```diff
@@ -119,58 +119,37 @@ public IParameterSet GetParametersForTemplate(IEngineEnvironmentSettings environ
-                (?<locale>
-                    [a-z]{2}
-                    (?:-[A-Z]{2})?
-                )
-                \."
-                + Regex.Escape(TemplateConfigFileName)
-                + "$",
-                RegexOptions.IgnorePatternWhitespace);
-
             IList<ITemplate> templateList = new List<ITemplate>();
             localizations = new List<ILocalizationLocator>();
 
-            foreach (IFile file in folder.EnumerateFiles("*" + TemplateConfigFileName, SearchOption.AllDirectories))
+            foreach (IFile file in folder.EnumerateFiles(TemplateConfigFileName, SearchOption.AllDirectories))
             {
-                if (string.Equals(file.Name, TemplateConfigFileName, StringComparison.OrdinalIgnoreCase))
-                {
-                    IFile hostConfigFile = file.MountPoint.EnvironmentSettings.SettingsLoader.FindBestHostTemplateConfigFile(file);
-
-                    if (TryGetTemplateFromConfigInfo(file, out ITemplate template, hostTemplateConfigFile: hostConfigFile))
-                    {
-                        templateList.Add(template);
-                    }
-
-                    continue;
-                }
+                IFile hostConfigFile = file.MountPoint.EnvironmentSettings.SettingsLoader.FindBestHostTemplateConfigFile(file);
 
-                Match localeMatch = localeFileRegex.Match(file.Name);
-                if (localeMatch.Success)
+                if (TryGetTemplateFromConfigInfo(file, out ITemplate template, hostTemplateConfigFile: hostConfigFile))
                 {
-                    string locale = localeMatch.Groups["locale"].Value;
+                    templateList.Add(template);
 
-                    if (TryGetLangPackFromFile(file, out ILocalizationModel locModel))
+                    IDirectory localizeFolder = file.Parent.DirectoryInfo("localize");
+                    if (localizeFolder != null && localizeFolder.Exists)
                     {
-                        ILocalizationLocator locator = new LocalizationLocator()
+                        foreach (IFile locFile in localizeFolder.EnumerateFiles("*" + LocalizationFileSuffix, SearchOption.AllDirectories))
                         {
-                            Locale = locale,
-                            MountPointUri = source.MountPointUri,
-                            ConfigPlace = file.FullPath,
-                            Identity = locModel.Identity,
-                            Author = locModel.Author,
-                            Name = locModel.Name,
-                            Description = locModel.Description,
-                            ParameterSymbols = locModel.ParameterSymbols
-                        };
-                        localizations.Add(locator);
+                            string locale = locFile.Name.Substring(0, locFile.Name.Length - LocalizationFileSuffix.Length);
+
+                            if (TryGetLangPackFromFile(locFile, out ILocalizationModel locModel))
```

Comment:
`TryGetLangPackFromFile` uses `SimpleConfigModel.LocalizationFromJObject(srcObject)` which doesn't have validation? Is it intentional? 
I remember we discussed having validation on installation, which seems not to be covered now. Will it be covered in next PR?

Reply:
We don't have validation at this point, because this part just extracts the values from the file. The validation happens later when we are comparing the template with the localization file prior to writing the cache.

This is probably a bit late for such validation, so I will add an earlier validation in the next PR with https://github.com/dotnet/templating/issues/3133.